pkg/messaging/pubsub/client.go

```diff
@@ -0,0 +1,60 @@
+package pubsub
+
+import (
+	"context"
+	"log"
+
+	"go.nanomsg.org/mangos/v3"
+	"go.nanomsg.org/mangos/v3/protocol"
+	"go.nanomsg.org/mangos/v3/protocol/sub"
+	_ "go.nanomsg.org/mangos/v3/transport/all"
+	"nodemon/pkg/entities"
+	"nodemon/pkg/messaging"
+)
+
+func subscribeToAlerts(socket protocol.Socket) error {
+	for alertType := range entities.AlertTypes {
+		err := socket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func StartMessagingClient(ctx context.Context, nanomsgURL string, bot messaging.Bot) error {
```

Comment:
All issues for `StartMessagingPairClient` also can be applied to this function. Fix them.